### ISSUE TITLE
Overfitting to training data! Missing a validation dataset?

### ISSUE THREAD

**Issue by PetrochukM** [Rayhane-mamah/Tacotron-2#34]
Hi There!

Been looking at the code, looks like there is no "test" or "validation" dataset; therefore, the loss decreasing only indicates that this code is memorizing the training data.

Thanks!

**Comment by gloriouskilka** [comment]
Well, if you run it manually and feed in unknown phrases, you can hear the output and make decision

**Comment by reidsanders** [comment]
I made a val split, and did find val loss slowly diverged from train loss (using m-ailabs elizabeth_klett, trained for 180k steps). Not sure what to do about it. Changes to the model take a long time to test (particularly when testing for overfitting). I'm looking for a good, very long english single speaker, but Google apparently trained on a 24.6 hour dataset, so this should be fine.

**Comment by Rayhane-mamah** [comment]
Hello, I am not late at all :)

it has been a while since I added a validation model to monitor overfit, like that we can stop training right before it happens. cheers!
